src/seismometer/controls/decorators.py

```diff
@@ -2,13 +2,79 @@
 """
 As for all of the utils subpackage, no other custom subpackages should be referenced.
 """
-import builtins
+import hashlib
+import logging
+import os
+import shutil
 from functools import wraps
+from inspect import signature
+from pathlib import Path
 from typing import Any, Callable
 
-from IPython.display import display
+from IPython.display import HTML
 from ipywidgets import Widget
 
+logger = logging.getLogger("seismometer")
+SEISMOMETER_CACHE_DIR = Path("./.seismometer_cache")
+SEISMOMETER_CACHE_ENABLED = os.getenv("SEISMOMETER_CACHE_ENABLED", "") != ""
+
+
+def disk_cached_html_segment(func: Callable[..., HTML]) -> HTML:
+    """
+    Caching decorator for HTML data
+
+    Parameters
+    ----------
+    func : Callable[..., HTML]
+        function that generates an HTML component
+
+    Returns
+    -------
+    HTML
+        displayable HTML component
+
+    Raises
+    ------
+    TypeError
+        If called with a function that does not annotate its return type as HTML, or that returns a non-HTML object.
+    """
+    sig = signature(func)
+    if sig.return_annotation != HTML:
+        raise TypeError("The function must return {HTML} object.")
+
+    if not SEISMOMETER_CACHE_ENABLED:
+        return func
+
+    @wraps(func)
+    def wrapped_func(*args, **kwargs):
+        arg_str = str((args, frozenset(kwargs.items())))
+        hash_object = hashlib.md5(arg_str.encode())
+        arg_hash = hash_object.hexdigest()
+
+        file_dir = Path("./.seismometer_cache") / "html" / func.__name__
```

Comment:
Should this use your global?

Reply:
FINC

Reply:
I think this was moved making the constants in controls/decorator.py unused. Can they be deleted in this file?  
I'm only seeing usage in the DiskCachedFunction now